### ISSUE TITLE
reproduce Bao et al on AAE2

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pie-document-level#367]
Finally, this is the command that should reproduce the setting from [Bao et al](https://aclanthology.org/2022.emnlp-main.713.pdf):
```
python src/train.py \
experiment=generic_pointer_network_end2end_re_past \
dataset=aae2_prepared \
dataset.create_validation_split.seed=12345 \
dataset.create_validation_split.test_size=0.15 \
dataset.input.revision=e035ff2c40c80448b48359ca7fe20a37db1b38d4 \
+dataset.input.conversion_method=connect_all \
+taskmodule.partition_layer_name=labeled_partitions \
+taskmodule.tokenizer_kwargs.strict_span_conversion=false \
model.base_model_config.lr=5e-5 \
callbacks.model_checkpoint.save_top_k=0 \
datamodule.batch_size=16 \
trainer=gpu \
trainer.min_epochs=75 \
trainer.max_epochs=75 \
trainer.deterministic=true \
test=true \
name=dataset-aae2/task-ner_re/bao \
seed=1,12,123,1234,12345 \
--multirun
```

Notes (differences to Bao et al in capital):
 - `seed=12345`: use a fixed seed for validation set creation. **THIS WILL RESULT IN A DIFFERENT VAL SET THAN BAO ET AL USED! THEY ALSO CREATE THE VAL SET ON THE PARAGRAPH LEVEL, WE DO IT ON THE DOCUMENT LEVEL.**
 - `test_size=0.15`: set val size to 15% (of train)
 - `conversion_method=connect_all`: have support/attack relations to the main claims in each paragraph that has a main claim (requires `dataset.input.revision=e035ff2c40c80448b48359ca7fe20a37db1b38d4`)
 - `partition_layer_name=labeled_partitions`: process each paragraph individually
 - `lr=5e-5`: same learning rate as Bao et al
 - `callbacks.model_checkpoint.save_top_k=0`: don't keep any intermediary model checkpoints, just keep the weights from from training on all epochs. By doing so, we do not rely on the `monitor_metric` ~~we implicitly use `monitor_metric=metric/binary_relations/micro/f1/val`.~~ **BAO USES `(metric/binary_relations/micro/f1/val + metric/labeled_spans/micro/f1/val) / 2` AS `monitor_metric` AND USES THE BEST CHECKPOINT REGARDING THAT FOR TESTING.**
 -  (see below). BAO USES monitor_metric
 - `(min|max)_epochs=75`: train for 75 epochs
 - `deterministic=true`: to make this again reproducible
 - train 5 runs with different. BAO TRAINED 10 RUNS.
 - we use `batch_size=16` (implicitly) to run on RTX2080Ti. BAO USES `batch_size=32`
 - **BAO USES A BucketSampler**

**Comment by ArneBinder** [comment]
This can be found here: https://github.com/ArneBinder/pie-document-level/pull/365#issuecomment-2640362014
